### PR TITLE
Bump CI disk image to Xcode 7.1.1

### PR DESCRIPTION
**Pull request by TimOliver**
Bumping our CI process to use Xcode 7.1 in order to take advantage of the new code features that were introduced to Objective-C in that version.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,3 +1,3 @@
 language: objective-c
-osx_image: xcode6.4
+osx_image: xcode7.1
```
Comment (jpsim) [inline .travis.yml]: Why not `xcode7.1.1`?

Comment (TimOliver) [inline .travis.yml]: Oh blimey! I didn't even notice that had come out.
No worries. I'll bump it to 7.1.1 then!

Comment (jpsim) [inline .travis.yml]: My bad, it's not available: http://docs.travis-ci.com/user/languages/objective-c/#Supported-OS-X-iOS-SDK-versions

You were correct in setting it to `xcode7.1` originally.

Comment (TimOliver) [inline .travis.yml]: Ah. 7.1.1 isn't actually supported on Travis. Only 7.1 or 7.2.

http://docs.travis-ci.com/user/languages/objective-c/

**Pull request merged**